feat(ZNTA-2297) Editor Activity Panel Backend

efloden opened the pull request:
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-2297

Implement REST API service to provide all project activity data to the React Editor.
Display this data (Translations, Comments, Revisions) in the frontend activity panel.
Add commenting functionality to the frontend activity panel

## QA
Activity panel should display all Translation History data (Translations, Comments, Revisions) 
Comment box should enforce a max length of 500 (same as the reject translation modals comment textarea) and clear itself after posting. The post comment button should be disabled while the comment textarea is empty.
Filter options should filter the history data by either all items, only comments or only revisions.
Copy buttons to the right of each item should copy the items comment or translation.
![activitypanel](https://user-images.githubusercontent.com/7971187/37321123-a6711708-26c2-11e8-8b83-2d8ed33267b0.png)

## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zanata/zanata-platform/729)
<!-- Reviewable:end -->

seanf commented:
Reviewed 26 of 28 files at r2.
Review status: all files reviewed at latest revision, 25 unresolved discussions.

---

*[server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java, line 2 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_UEmU0tFDo7LZEhDs:-L6_UEmViV-_WnZ3QZrC:b-k8o9kb) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java#L2)):*
> ```Java
> /*
>  * Copyright 2010, Red Hat, Inc. and individual contributors
> ```

wrong year

---

*[server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java, line 41 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_UOAiY6_P9OUCviZu:-L6_UOAiY6_P9OUCviZv:b-ekjskq) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java#L41)):*
> ```Java
>  */
> @RequestScoped
> @Named("editor.transUnitHistoryService")
> ```

Unless you're calling this from a JSF page, it shouldn't need a name.

---

*[server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java, line 71 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_VkUwcEKaphoq79Ie:-L6_VkUwcEKaphoq79If:b-csbb92) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java#L71)):*
> ```Java
>         if (transUnitId == null) {
>             return Response.status(Response.Status.BAD_REQUEST)
>                     .entity("Null transUnitId supplied")
> ```

That's a lot of error-handling boilerplate (not to mention four test cases in the test class) for a function with three lines of real logic. 

This is a general problem, but it would be good to create a general function which checks for missing parameters, so that we can just pass in a list of param names and values (perhaps Pairs), and that function would generate the BAD_REQUEST response for us.

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java, line 72 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_X0sDNAVVH1FAPqcD:-L6_X0sDNAVVH1FAPqcE:b-9j1mg0) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java#L72)):*
> action.getWorkspaceId()

You should extract this to a local variable and reuse it.

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java, line 117 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_XSYvCnJJJlQKi3uQ:-L6_XSYvCnJJJlQKi3uR:b-w1u46t) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java#L117)):*
> ```Java
>                     HTextFlow hTextFlow, HLocale hLocale) {
>         TransHistoryItem latest = null;
>         if (hTextFlowTarget != null) {
> ```

You shouldn't need to repeat this check if you call getLatest inside the not-null check of getTranslationHistory.

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java, line 140 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_YIzVwEcgTiY4MEwf:-L6_YIzVwEcgTiY4MEwg:b-h56ath) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java#L140)):*
> ```Java
>                         hLocale.getLocaleId());
>         return Lists.transform(hComments, input -> new ReviewComment(
>                 new ReviewCommentId(input.getId()),
> ```

:+1: for converting to lambda.

---

*[server/services/src/test/java/org/zanata/rest/editor/service/TransUnitHistoryServiceTest.java, line 14 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_Ybgli4AGvoeAdnqP:-L6_Ybgli4AGvoeAdnqQ:b35ucrg) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/test/java/org/zanata/rest/editor/service/TransUnitHistoryServiceTest.java#L14)):*
> ```Java
> import static org.assertj.core.api.Assertions.assertThat;
> 
> public class TransUnitHistoryServiceTest {
> ```

Where's the test for the successful case?

---

*[server/services/src/test/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandlerTest.java, line 295 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_ZBkwNKrTMpxtPbZv:-L6_ZBkwNKrTMpxtPbZw:b-npkryk) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/test/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandlerTest.java#L295)):*
> ```Java
> 
>         List<ReviewComment> result =
>                 handler.getReviewComments(action.getTransUnitId(), hLocale);
> ```

Why did this test need to change? You can still pass in an action, can't you?

Why don't you split out two tests? One creates a GetTranslationHistoryAction and passes it in as before; the other shouldn't even need a GetTranslationHistoryAction, but will use the same method as your new REST service.

---

*[server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap, line 37 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6__0Y37q2l4mX3OcCb:-L6__0Y37q2l4mX3OcCc:b2cpqmu) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap#L37)):*
> ```
>         href="/profile/view/damason"
>       >
>         damason
> ```

Was this change intentional? Perhaps we should include both names, something like: `David Mason (damason)`

---

*[server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap, line 44 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6__2Z5lw7Krf2X7fcs:-L6__2Z5lw7Krf2X7fct:b6zi7m) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap#L44)):*
> ```
>       >
>         <span>
>           approved this translation
> ```

Was this change intentional?

---

*[server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap, line 149 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_g191h3euaAnKggWX:-L6_g191h3euaAnKggWY:bfo6ptx) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap#L149)):*
> ```
>         damason
>       </a>
>        has commented on this translation
> ```

Commented on this translation? Are we showing which translation that is?

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/ActivityFeedItem.story.js, line 41 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6__IB0rsG5f3fcoek6:-L6__IB0rsG5f3fcoek7:bgqje2v) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/ActivityFeedItem.story.js#L41)):*
> ```JavaScript
>     .add('comment', () => (
>       <ActivityFeedItem
>         type='comment'
> ```

Why did the quotes change? I think TSX insists on double quotes for attributes.

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js, line 59 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_a4Dp0BbNHwORhPTj:-L6_a4Dp0BbNHwORhPTk:b-jk4dod) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js#L59)):*
> ```JavaScript
>       CRITICAL
>     ]),
>     status: PropTypes.oneOf(['translated', 'needswork', 'approved', 'rejected',
> ```

Why the change?

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js, line 134 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_aHoOzAkgAhejPAjI:-L6_aHoOzAkgAhejPAjJ:b-awpqjk) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js#L134)):*
> ```JavaScript
>       /*
>        * TODO: Source and display the criteria and priority of a rejection:
>        * defaultMessage -> "for the reason: {criteria} - {priority} priority."
> ```

Why the change? Also, this comment doesn't need to be up here; it could be right next to defaultMessage. (You can have ordinary JS comments in JSX between attributes.)

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js, line 189 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6__qmydxvT9LdBwGBt:-L6__qmydxvT9LdBwGBu:bbomnvz) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js#L189)):*
> ```JavaScript
>         // Nested formatted messages are needed to highlight part of the message
>         const fuzzyRevision = (
>           <span className='u-textUnsure'>
> ```

More JSX quote changes throughout this file...

---

*[server/zanata-frontend/src/app/editor/components/CommentBox/index.js, line 33 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_cDrhBB-XqOaojozi:-L6_cDrhBB-XqOaojozj:b6ekuax) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/CommentBox/index.js#L33)):*
> ```JavaScript
>   }
> 
>   defaultState = {
> ```

Can we make this private?

---

*[server/zanata-frontend/src/app/editor/components/CommentBox/index.js, line 61 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_cGRHWADjf_6Nf1i3:-L6_cGRHWADjf_6Nf1i4:b-bhctkb) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/CommentBox/index.js#L61)):*
> ```JavaScript
>           </ControlLabel><br />
>           <FormControl componentClass='textarea'
>             placeholder='...' onChange={this.setCommentText} />
> ```

More quote changes...

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.js, line 18 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_c_wgAI2FH-FWtaPV:-L6_c_wgAI2FH-FWtaPW:b-cdcp34) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/ActivityTab.js#L18)):*
> ```JavaScript
>  * selected={selectedLanguages} />
>  */
> const ActivityTab = ({
> ```

So you turned a stateful component into a stateless one? Excellent.

But I'd feel more confident about this change if this file were type checked. It might be worth renaming to a .ts file and having this function typed as `React.SFC` (which will probably fix an error about ActivityTab.propTypes below). We can try this together on Monday if you like.

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.js, line 49 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_dZqHULdIrLWgXjOo:-L6_dZqHULdIrLWgXjOp:b2jfnd1) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/ActivityTab.js#L49)):*
> ```JavaScript
>     // TODO damason define type for status
>     status: PropTypes.oneOf(['translated', 'needswork', 'approved',
>     'rejected', 'untranslated']),
> ```

It would be good to add such a PropType to `phrase.ts` and fix this TODO.

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.js, line 67 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_d1npt2KkfSi9VX2n:-L6_d1npt2KkfSi9VX2o:bd1vjv0) ([raw file](https://github.com/zanata/zanata-platform/blob/3028f9dde0ba14a394fecd6044d164c26c15ceca/server/zanata-frontend/src/app/editor/containers/ActivityTab.js#L67)):*
> ```JavaScript
> function mapStateToProps (_state) {
>   // Dummy data. Kept for structure reference.
>   // FIXME delete when the component is wired into the app.
> ```

I guess you wired it in?

---

*[server/zanata-frontend/src/app/editor/containers/GlossaryTab.js, line 117 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_eZYzI8aS6Udt3cJ5:-L6_eZYzI8aS6Udt3cJ6:b-wx4dqt) ([raw file](https://github.com/zanata/zanata-platform/blob/3028f9dde0ba14a394fecd6044d164c26c15ceca/server/zanata-frontend/src/app/editor/containers/GlossaryTab.js#L117)):*
> ```JavaScript
>     const directionClass = isRTLSource ? 'rtl' : 'ltr'
>     return (
>       <Tab eventKey={eventKey} title="">
> ```

What's this change about?

---

*[server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js, line 1 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_ekU3R-SjFFBme5ID:-L6_ekU3R-SjFFBme5IE:b-dt6ush) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js#L1)):*
> ```JavaScript
> // @ts-nocheck
> ```

Given the size of the change, it would be good to get the types to check in this file (and/or rename to .ts).

---

*[server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js, line 174 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_f0WIbPfegm_2CBu2:-L6_f0WIbPfegm_2CBu3:b5uynkm) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js#L174)):*
> ```JavaScript
>           name: value.commenterName,
>           imageUrl:
>             'https://gravatar.com/avatar/a0c33fb16389ac6c3d7034efb1f3f305'
> ```

What's this?

---

*[server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js, line 193 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_f1ny24khFWZRnqXU:-L6_f1ny24khFWZRnqXV:b-1ec2i5) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js#L193)):*
> ```JavaScript
>           name: historyItem.modifiedBy,
>           imageUrl:
>             'https://gravatar.com/avatar/a0c33fb16389ac6c3d7034efb1f3f305'
> ```

And this?

---

*[server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js, line 207 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_f3935lFhhXD7AvYY:-L6_f3935lFhhXD7AvYZ:b1p0oul) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js#L207)):*
> ```JavaScript
>         name: latestHistoryItem.modifiedBy,
>         imageUrl:
>           'https://gravatar.com/avatar/a0c33fb16389ac6c3d7034efb1f3f305'
> ```

And this one?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

efloden commented:
Review status: 18 of 27 files reviewed at latest revision, 24 unresolved discussions.

---

*[server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java, line 2 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_UEmU0tFDo7LZEhDs:-L6nI4RZVwrXyUghwhmx:bvnrpcm) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java#L2)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

wrong year
</blockquote></details>

updated

---

*[server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java, line 41 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_UOAiY6_P9OUCviZu:-L6nI7x6r4XnH7Mh2FFM:byjv5b3) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java#L41)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Unless you're calling this from a JSF page, it shouldn't need a name.
</blockquote></details>

removed the name

---

*[server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java, line 71 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_VkUwcEKaphoq79Ie:-L6onX1VL2MFvKVPQ81N:b-ugw1uz) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/rest/editor/service/TransUnitHistoryService.java#L71)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

That's a lot of error-handling boilerplate (not to mention four test cases in the test class) for a function with three lines of real logic. 

This is a general problem, but it would be good to create a general function which checks for missing parameters, so that we can just pass in a list of param names and values (perhaps Pairs), and that function would generate the BAD_REQUEST response for us.
</blockquote></details>

Refactored to use a checkParams function in RestUtils

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java, line 72 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_X0sDNAVVH1FAPqcD:-L6nUKNJbidUPzxX2ywT:bt3mixd) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java#L72)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

> action.getWorkspaceId()

You should extract this to a local variable and reuse it.
</blockquote></details>

Extracted to local variable

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java, line 117 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_XSYvCnJJJlQKi3uQ:-L6nUdl1Cdx5baZKfMCO:btgdqus) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java#L117)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

You shouldn't need to repeat this check if you call getLatest inside the not-null check of getTranslationHistory.
</blockquote></details>

Looks like I had to check if hTextFlow or hTextFlowTarget are null before continuing to avoid null pointer errors for other functions as well, move these checks up into getTranslationHistory

---

*[server/services/src/test/java/org/zanata/rest/editor/service/TransUnitHistoryServiceTest.java, line 14 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_Ybgli4AGvoeAdnqP:-L6otGXSTPFsKq0tV_-e:b-mvpagh) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/test/java/org/zanata/rest/editor/service/TransUnitHistoryServiceTest.java#L14)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Where's the test for the successful case?
</blockquote></details>

Implemented the test for a successful case by mocking historyHandler correctly (not overwriting it)

---

*[server/services/src/test/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandlerTest.java, line 295 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_ZBkwNKrTMpxtPbZv:-L6ot_NKIhoWIDIsiFGD:b-jyezqd) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/services/src/test/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandlerTest.java#L295)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Why did this test need to change? You can still pass in an action, can't you?

Why don't you split out two tests? One creates a GetTranslationHistoryAction and passes it in as before; the other shouldn't even need a GetTranslationHistoryAction, but will use the same method as your new REST service.
</blockquote></details>

I changed the getReviewComments function to accept a transUnitId and hLocale

---

*[server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap, line 37 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6__0Y37q2l4mX3OcCb:-L6nV3wl_Tl4qpeigpPM:b-j2hrh7) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap#L37)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Was this change intentional? Perhaps we should include both names, something like: `David Mason (damason)`
</blockquote></details>

From what I can see in the database we do not store a full name for users

---

*[server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap, line 44 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6__2Z5lw7Krf2X7fcs:-L6nVHFWB01MQQBDugYb:b-s09v9c) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap#L44)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Was this change intentional?
</blockquote></details>

Yes, changed to reflect that the messages relate to the currently selected transunit

---

*[server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap, line 149 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_g191h3euaAnKggWX:-L6nVYgTTgMQHBtS_tll:b-rkkpf0) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap#L149)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Commented on this translation? Are we showing which translation that is?
</blockquote></details>

Yes, it is a translation of the currently selected transunit

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/ActivityFeedItem.story.js, line 41 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6__IB0rsG5f3fcoek6:-L6olpphuNkFHWYVvugk:b-t7p6m8) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/ActivityFeedItem.story.js#L41)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Why did the quotes change? I think TSX insists on double quotes for attributes.
</blockquote></details>

This code was not being used previously, the eslint rule about single quotes will cause the watch task to fail now that this code is executed

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js, line 59 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_a4Dp0BbNHwORhPTj:-L6omCK65lY28NXdoJx0:bzdsd2x) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js#L59)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Why the change?
</blockquote></details>

The actual data structure for status used in the app is 'needswork', I think this code predates that implementation

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js, line 134 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_aHoOzAkgAhejPAjI:-L6omgH30DOiiAg7LqYm:b-f3621t) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js#L134)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Why the change? Also, this comment doesn't need to be up here; it could be right next to defaultMessage. (You can have ordinary JS comments in JSX between attributes.)
</blockquote></details>

We don't have saving a priority selection implemented for translation rejections yet, so I disabled this display for now (implementing priority selection will include database table updates)

---

*[server/zanata-frontend/src/app/editor/containers/GlossaryTab.js, line 117 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_eZYzI8aS6Udt3cJ5:-L6o7HNWFg3kBcrbVNnB:bciqtdc) ([raw file](https://github.com/zanata/zanata-platform/blob/3028f9dde0ba14a394fecd6044d164c26c15ceca/server/zanata-frontend/src/app/editor/containers/GlossaryTab.js#L117)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

What's this change about?
</blockquote></details>

The TabGroup component requires Tab to be direct children components

---

*[server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js, line 174 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_f0WIbPfegm_2CBu2:-L6s_APjDGqNCUPgwadg:bejyfz6) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/TranslationInfoPanel/index.js#L174)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

What's this?
</blockquote></details>

These are avatar URLs for the users attached to a given history item or revision comment, currently we aren't providing the information required to build their gravatar URL with the transunithistory response, and avatars for comments/revisions are not present in the GWT editor, so will remove the avatars 

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

efloden commented:
Review status: 17 of 28 files reviewed at latest revision, 6 unresolved discussions.

---

*[server/zanata-frontend/src/app/editor/components/CommentBox/index.js, line 33 at r2](https://beta.reviewable.io/reviews/zanata/zanata-platform/729#-L6_cDrhBB-XqOaojozi:-L6xdp1j0ykGrVJiui1b:ba2ri70) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/CommentBox/index.js#L33)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Can we make this private?
</blockquote></details>

Converted to .tsx file with private methods

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

efloden commented:
Review status: 24 of 30 files reviewed at latest revision, 5 unresolved discussions.

---

*[server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap, line 37 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6__0Y37q2l4mX3OcCb:-L6yhtBoIycEHYIvf1g8:b-l820m0) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/.storybook-editor/__snapshots__/storyshots-editor.test.js.snap#L37)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

We do have HPerson.name in the database. It is intended to be the full name, or at least the user's "human-readable" name.
</blockquote></details>

Okay, that makes sense - I will need to change the java service to return both the HPerson.name and HAccount.name for revisions and comments to display these correctly then

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js, line 59 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_a4Dp0BbNHwORhPTj:-L6yiTZ0GTVLuc-3cHu1:b-826hh7) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js#L59)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Oh, you mean this was dead code until now, and had a bug in the PropTypes?
</blockquote></details>

Yes, updated this to also use the statuses tuple type now

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js, line 64 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6yTep195Z9zpkReOzr:-L7761f4z7z4RkxYFn8E:blzlobv) ([raw file](https://github.com/zanata/zanata-platform/blob/3028f9dde0ba14a394fecd6044d164c26c15ceca/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js#L64)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

We're probably going to need both of these names. I think name is `HPerson.name`, and username is `HAccount.username`. That would the intention, anyway.
</blockquote></details>

I see, it looks like the revision items have the HAccount.username attached while the comments have a HPerson.name, I will need to change the backend service to return both for each item to handle this properly. Should I attempt this in this 

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.js, line 18 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_c_wgAI2FH-FWtaPV:-L77N_dARFGnuk6mAuB4:bph24m) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/ActivityTab.js#L18)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

So you turned a stateful component into a stateless one? Excellent.

But I'd feel more confident about this change if this file were type checked. It might be worth renaming to a .ts file and having this function typed as `React.SFC` (which will probably fix an error about ActivityTab.propTypes below). We can try this together on Monday if you like.
</blockquote></details>

Converted to a typescript SFC component using new and existing types

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.js, line 49 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_dZqHULdIrLWgXjOo:-L77H_e3kxU29Bm1ZgKc:b-ysuofk) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/ActivityTab.js#L49)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

It would be good to add such a PropType to `phrase.ts` and fix this TODO.
</blockquote></details>

Now using the Status type from phrase.ts

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

seanf commented:
Reviewed 6 of 7 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js, line 59 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_a4Dp0BbNHwORhPTj:-L77ca_xqLnvmmo8A4C9:b-8fx2u7) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js#L59)):*
<details><summary><i>Previously, efloden (Earl Floden) wrote…</i></summary><blockquote>

Yes, updated this to also use the statuses tuple type now
</blockquote></details>

:+1: 

---

*[server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js, line 64 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6yTep195Z9zpkReOzr:-L77gqBBWCNmTiGpqD_S:b-kr60b7) ([raw file](https://github.com/zanata/zanata-platform/blob/3028f9dde0ba14a394fecd6044d164c26c15ceca/server/zanata-frontend/src/app/editor/components/ActivityFeedItem/index.js#L64)):*
<details><summary><i>Previously, efloden (Earl Floden) wrote…</i></summary><blockquote>

I see, it looks like the revision items have the HAccount.username attached while the comments have a HPerson.name, I will need to change the backend service to return both for each item to handle this properly. Should I attempt this in this 
</blockquote></details>

Yes, I think that would be best, thanks Earl.

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.js, line 18 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L6_c_wgAI2FH-FWtaPV:-L77d0vSCiXGLEG7gtSK:b-8fx2u7) ([raw file](https://github.com/zanata/zanata-platform/blob/386a053b7cc101b0770d9869acc1a3e85bf0d0bb/server/zanata-frontend/src/app/editor/containers/ActivityTab.js#L18)):*
<details><summary><i>Previously, efloden (Earl Floden) wrote…</i></summary><blockquote>

Converted to a typescript SFC component using new and existing types
</blockquote></details>

:+1: 

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx, line 23 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L77fBxLoj9w8nE8g1D3:-L77fBxLoj9w8nE8g1D4:bgr14t5) ([raw file](https://github.com/zanata/zanata-platform/blob/3f87c62d781efdc33f5b6766e770bc70c4dc7aac/server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx#L23)):*
> ```XML
>  * TODO: Implement or remove LanguageSelectList
>  */
> const ActivityTab: React.SFC<ActivityTabProps> = ({
> ```

What's happened to the propTypes for ActivityTab? Ref: https://github.com/zanata/zanata-platform/blob/3028f9dde0ba14a394fecd6044d164c26c15ceca/server/zanata-frontend/src/app/editor/containers/ActivityTab.js

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx, line 39 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L77g8QAr9YjC5my2IeH:-L77g8QAr9YjC5my2IeI:b-5bccg6) ([raw file](https://github.com/zanata/zanata-platform/blob/3f87c62d781efdc33f5b6766e770bc70c4dc7aac/server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx#L39)):*
> ```XML
>           selected={selectedActivites} />
>         {/* <LanguageSelectList selectItem={selectLanguageFilter}
>           *  selected={selectedLanguages} /> */}
> ```

This component was dead code too, before, right? Thus the LanguageSelectList part was never really used?

An activity list seems like a weird place for a language selector anyway. Was the idea to filter activity by language?

---

*[server/zanata-frontend/src/app/editor/utils/activity-util.ts, line 30 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L77dyM7ZXtz4zz9COWP:-L77dyM7ZXtz4zz9COWQ:b-q8v0sn) ([raw file](https://github.com/zanata/zanata-platform/blob/3f87c62d781efdc33f5b6766e770bc70c4dc7aac/server/zanata-frontend/src/app/editor/utils/activity-util.ts#L30)):*
> ```XML
>     id: ALL,
>     icon: 'clock',
>     label: 'All'
> ```

Is there some sort of interface we could create or use here?  These elements seem to be different from ActivityItem above. It's unclear what this is really a list of.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

efloden commented:
Review status: 25 of 30 files reviewed at latest revision, 5 unresolved discussions.

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx, line 23 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L77fBxLoj9w8nE8g1D3:-L77iyhejiK7E7sUcpPY:bcwz0v) ([raw file](https://github.com/zanata/zanata-platform/blob/3f87c62d781efdc33f5b6766e770bc70c4dc7aac/server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx#L23)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

What's happened to the propTypes for ActivityTab? Ref: https://github.com/zanata/zanata-platform/blob/3028f9dde0ba14a394fecd6044d164c26c15ceca/server/zanata-frontend/src/app/editor/containers/ActivityTab.js
</blockquote></details>

I have converted them into a typescript interface now, the eventKey is handled in the parent component now (because of the bootstrap Tabs needing to be directly nested) , selectLanguageFilter and selectedLanguages were dead code, more on that below

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx, line 39 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L77g8QAr9YjC5my2IeH:-L77jW1uzwdgS0KXl2nz:bhzfesz) ([raw file](https://github.com/zanata/zanata-platform/blob/3f87c62d781efdc33f5b6766e770bc70c4dc7aac/server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx#L39)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

This component was dead code too, before, right? Thus the LanguageSelectList part was never really used?

An activity list seems like a weird place for a language selector anyway. Was the idea to filter activity by language?
</blockquote></details>

Yes this was not being used, I think the intention might have been to filter by other languages if the state was holding all the translation history data for all languages, but that isn't really the case with the current implementation. We do have the option to change project language at the top of the page already.

---

*[server/zanata-frontend/src/app/editor/utils/activity-util.ts, line 30 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L77dyM7ZXtz4zz9COWP:-L77lj-591raugq_nCSf:bx2se35) ([raw file](https://github.com/zanata/zanata-platform/blob/3f87c62d781efdc33f5b6766e770bc70c4dc7aac/server/zanata-frontend/src/app/editor/utils/activity-util.ts#L30)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Is there some sort of interface we could create or use here?  These elements seem to be different from ActivityItem above. It's unclear what this is really a list of.
</blockquote></details>

These are used for the display of the activity item SelectButtons (all, comments or updates), it looks like they are pulled out here to allow for re-use of a component for a list of language select items. I've tried to make a more intuitive interface for these

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

seanf commented:
Reviewed 5 of 5 files at r6.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx, line 23 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L77fBxLoj9w8nE8g1D3:-L788-y3ZlmaDRqV9Yan:bgm83k2) ([raw file](https://github.com/zanata/zanata-platform/blob/3f87c62d781efdc33f5b6766e770bc70c4dc7aac/server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx#L23)):*
<details><summary><i>Previously, efloden (Earl Floden) wrote…</i></summary><blockquote>

I have converted them into a typescript interface now, the eventKey is handled in the parent component now (because of the bootstrap Tabs needing to be directly nested) , selectLanguageFilter and selectedLanguages were dead code, more on that below
</blockquote></details>

I understand about the props which went away, but I thought the idea was that every one of our components should have PropTypes for runtime validation (since we can't be sure every user of a component is fully typed).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

efloden commented:
Review status: 29 of 30 files reviewed at latest revision, 3 unresolved discussions.

---

*[server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx, line 23 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L77fBxLoj9w8nE8g1D3:-L78AgVKsgaUYnx8OQUW:b-hqask3) ([raw file](https://github.com/zanata/zanata-platform/blob/3f87c62d781efdc33f5b6766e770bc70c4dc7aac/server/zanata-frontend/src/app/editor/containers/ActivityTab.tsx#L23)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

I understand about the props which went away, but I thought the idea was that every one of our components should have PropTypes for runtime validation (since we can't be sure every user of a component is fully typed).
</blockquote></details>

Oh right, my mistake, I have reinstated the proptypes for this component now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

seanf commented:
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

seanf commented:
Reviewed 8 of 8 files at r8.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/ReviewComment.java, line 19 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7MbfbUCoQ9DY96tL75:-L7MbfbUCoQ9DY96tL76:b-7feuwv) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/ReviewComment.java#L19)):*
> ```Java
>     private String comment;
>     private String commenterName;
>     private String accountName;
> ```

For consistency with the rest of the code, could you please make this `username`?

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/ReviewComment.java, line 52 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7MpVhkipPTSIdD3n-1:-L7MpVhkipPTSIdD3n-2:b-sy1poh) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/ReviewComment.java#L52)):*
> ```Java
> 
>     public ReviewComment setAccountName(String accountName) {
>         this.accountName = accountName;
> ```

Please move this to the constructor instead of making the class mutable.

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java, line 22 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7Mg9x3xnkL2nnteXaK:-L7Mg9x3xnkL2nnteXaL:b-dpd9oa) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java#L22)):*
> ```Java
>     private List<String> contents;
>     private ContentState status;
>     private String modifiedBy;
> ```

/** account username */

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java, line 23 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7MgDfvj8BgrHsb4XBD:-L7MgDfvj8BgrHsb4XBE:bx8jod9) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java#L23)):*
> ```Java
>     private ContentState status;
>     private String modifiedBy;
>     private String modifiedByUser = "";
> ```

maybe modifiedByPersonName

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java, line 95 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7Mfw0zWdscLV6l92tW:-L7Mfw0zWdscLV6l92tX:bbcvvpk) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java#L95)):*
> ```Java
>     }
> 
>     public TransHistoryItem setModifiedByUser(String username) {
> ```

param should be personName

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java, line 149 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7Mqk0KHB5s897_DZK1:-L7Mqk0KHB5s897_DZK2:b-kv52cz) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java#L149)):*
> && lastModifiedBy.hasAccount()
Shouldn't need this check.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

efloden commented:
Review status: 25 of 35 files reviewed at latest revision, 6 unresolved discussions.

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/ReviewComment.java, line 19 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7MbfbUCoQ9DY96tL75:-L7MxOMVoM5mjlMyxXfJ:b2xznrb) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/ReviewComment.java#L19)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

For consistency with the rest of the code, could you please make this `username`?
</blockquote></details>

Refactored to usename

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/ReviewComment.java, line 52 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7MpVhkipPTSIdD3n-1:-L7NIO1ZKGyj7fcurY2e:b-nwkkjk) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/ReviewComment.java#L52)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Please move this to the constructor instead of making the class mutable.
</blockquote></details>

Moved to constructor and removed setter method

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java, line 22 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7Mg9x3xnkL2nnteXaK:-L7MxUotTAV9GtIzTwCn:b-8b11j9) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java#L22)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

/** account username */
</blockquote></details>

Added

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java, line 23 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7MgDfvj8BgrHsb4XBD:-L7MxYdHKKbVLH9vbfp0:bc95e17) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java#L23)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

maybe modifiedByPersonName
</blockquote></details>

Refactored to modifiedByPersonName

---

*[server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java, line 95 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7Mfw0zWdscLV6l92tW:-L7Mxa2IBUMdPt5S1Mt-:bo1mzu3) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/gwt-shared/src/main/java/org/zanata/webtrans/shared/model/TransHistoryItem.java#L95)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

param should be personName
</blockquote></details>

Refactored to personName

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java, line 149 at r8](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7Mqk0KHB5s897_DZK1:-L7Mxl0vyS2lQr3lCEql:bad246j) ([raw file](https://github.com/zanata/zanata-platform/blob/094e45e256feaaeb26130e11d103e70d2f7a6971/server/services/src/main/java/org/zanata/webtrans/server/rpc/GetTranslationHistoryHandler.java#L149)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

> && lastModifiedBy.hasAccount()
Shouldn't need this check.
</blockquote></details>

Removed this check

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

seanf commented:
Reviewed 10 of 10 files at r9.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/AddReviewCommentHandler.java, line 134 at r9](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7NoNvz-AO9-Sv0sF5u:-L7NoNw-G4j9nn73f5_-:bv3pfz2) ([raw file](https://github.com/zanata/zanata-platform/blob/2b38b5c3571a477f0268547b2c4e2677a33c3560/server/services/src/main/java/org/zanata/webtrans/server/rpc/AddReviewCommentHandler.java#L134)):*
> ```Java
>         return new ReviewComment(new ReviewCommentId(hComment.getId()),
>                 hComment.getComment(), hComment.getCommenterName(),
>                 hComment.getCommenterUsername(),
> ```

We should try to be consistent about the ordering. I would suggest username then personName.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

efloden commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/AddReviewCommentHandler.java, line 134 at r9](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7NoNvz-AO9-Sv0sF5u:-L7NuDw-pj8e241FmUL3:blq2a1q) ([raw file](https://github.com/zanata/zanata-platform/blob/2b38b5c3571a477f0268547b2c4e2677a33c3560/server/services/src/main/java/org/zanata/webtrans/server/rpc/AddReviewCommentHandler.java#L134)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

We should try to be consistent about the ordering. I would suggest username then personName.
</blockquote></details>

Do you mean to swap the positions of username and commenterName in the ReviewComment constructor?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729)*
<!-- Sent from Reviewable.io -->

seanf commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 4 files at r10.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[server/services/src/main/java/org/zanata/webtrans/server/rpc/AddReviewCommentHandler.java, line 134 at r9](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-L7NoNvz-AO9-Sv0sF5u:-L7O29G7rOjZmKXeiBFv:belfa9p) ([raw file](https://github.com/zanata/zanata-platform/blob/2b38b5c3571a477f0268547b2c4e2677a33c3560/server/services/src/main/java/org/zanata/webtrans/server/rpc/AddReviewCommentHandler.java#L134)):*
<details><summary><i>Previously, efloden (Earl Floden) wrote…</i></summary><blockquote>

Do you mean to swap the positions of username and commenterName in the ReviewComment constructor?
</blockquote></details>

Yes, that's it. Good.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/729#-:-L7O2Ur8eZsM0vy9O0Wi:bnfp4nl)*
<!-- Sent from Reviewable.io -->

spathare21 commented:
@efloden looks like that post comment button are not working, haven't seen added comment, 
Also, the text box for the comment it going out of a page.

efloden commented:
@spathare21 Fixed the padding style on the text box.
The comment was being added to the database but I was not updating the UI with the new comment on success. Thanks for catching that, should be fixed now.

spathare21 commented:
@efloden A long string comment is not wrapping correctly 

![screenshot from 2018-03-16 15-02-28](https://user-images.githubusercontent.com/23230884/37513452-21f8834c-292b-11e8-9d9e-86d39bb53179.png)

efloden commented:
@spathare21 Comment wrapping should be fixed
